### PR TITLE
#532 Changed Typographies to Form Labels

### LINKED ISSUE #532: [Tech Debt] - Change MUI Labels to Form Labels
### Description

Currently in a couple places, we use typographies to label input boxes rather than formal labels and form controls like we should. We wanna use form labels because they look better and are easier to change how we want. We want to update all the spots listed to use FormControls and Form Labels (see the CreateProjectFormView).

Project Edit - Project edit uses this built in label prop mui has, but I don't like it, it's prone to weird bugs and isn't worth the trouble imo, so we're switching to a formcontrol with form label on all the inputs in this file. If you have more questions about that, just slack me in a channel - Reid
Work Package Edit - same as Project Edit

### Acceptance Criteria

All the listed forms use the form label instead of typographies. 

### Proposed Solution

See CreateProjectFormView for examples of how to do this.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by zmar234**
## Changes

In ProjectEditDetails.tsx I added form labels and form controls to each element that takes in data on the edit page and gave them appropriately named placeholder text.

In WorkPackageEditDetails.tsx I did the same as above except I did not need a form control in some areas. 

## Notes

I had a lot of trouble getting the elements in ProjectEditDetails.tsx to stay aligned when I added form controls. I have no clue why it was this file specifically when I had little issue with WorkPackageEditDetails.tsx. While it works fine right now I believe it should be revisited in the future (preferably by someone with more experience than me) to look at and clean it up if needed.

## Screenshots

image
image
image
image

Closes #532

### CODE REVIEW

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 1]
```diff
@@ -17,90 +17,108 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
         <Grid item xs={12} md={6} sx={{ mt: 2, mb: 1 }}>
-          <ReactHookTextField
-            name="budget"
-            control={control}
-            type="number"
-            label="Budget"
-            sx={{ width: 3 / 10 }}
-            icon={<AttachMoneyIcon />}
-            errorMessage={errors.budget}
-          />
```
Comment (nezamj03) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:37]: Screen Shot 2023-02-06 at 12 24 43 AM


This is what your branch looks like for me. I'm not sure why there's a difference, but maybe try going fullscreen/ getting out of fullscreen and see what it looks like.

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 2]
```diff
@@ -26,13 +26,16 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
...
+        <FormControl>
+          <FormLabel>Status</FormLabel>
+          <TextField select onChange={onChange} value={value}>
+            {statuses.map((t) => (
+              <MenuItem key={t} value={t}>
+                {t}
+              </MenuItem>
+            ))}
+          </TextField>
+        </FormControl>
```
Comment (nezamj03) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:38]: Screen Shot 2023-02-06 at 12 26 35 AM


less important, but when i shrink the screen a little, the formatting gets a little weird

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 3]
```diff
@@ -17,90 +17,108 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
+            <FormLabel>Budget</FormLabel>
+            <ReactHookTextField
+              name="budget"
+              control={control}
+              type="number"
+              placeholder="Enter budget..."
+              icon={<AttachMoneyIcon />}
+              errorMessage={errors.budget}
+            />
+          </FormControl>
```
Comment (nezamj03) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:41]: Screen Shot 2023-02-06 at 12 24 43 AM


This is what your branch looks like for me. I'm not sure why there's a difference, but maybe try going fullscreen/ getting out of fullscreen and see what it looks like.

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx (hunk 2 above)
Comment (nezamj03) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:38]: Screen Shot 2023-02-06 at 12 26 35 AM


less important, but when i shrink the screen a little, the formatting gets a little weird

**Review by nezamj03 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 4]
```diff
@@ -41,11 +44,12 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
     <PageBlock title="Work Package Details">
       <Grid container spacing={1}>
         <Grid item xs={12} md={6} sx={{ mt: 2, mb: 1 }}>
+          <FormLabel>Work Package Name</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:47]: should be in a form control

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx (hunk 2 above)
Comment (RChandler234) [reply src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:38]: you'll probably have to change the size of the grid item for date when it's extra small using the xs prop

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 5]
```diff
@@ -56,8 +60,8 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
             rules={{ required: true }}
             render={({ field: { onChange, value } }) => (
               <>
+                <FormLabel>Start Date (YYYY-MM-DD)</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:63]: FormControl

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 6]
```diff
@@ -77,13 +81,16 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
...
             render={({ field: { onChange, value } }) => (
-              <TextField select onChange={onChange} value={value} label="Project Lead" fullWidth>
-                {users.map((t) => (
-                  <MenuItem key={t.userId} value={t.userId}>
-                    {fullNamePipe(t)}
-                  </MenuItem>
-                ))}
-              </TextField>
+              <>
+                <FormLabel>Project Lead</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:85]: FormControl

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 7]
```diff
@@ -93,25 +100,31 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
...
             render={({ field: { onChange, value } }) => (
-              <TextField select onChange={onChange} value={value} label="Project Manager" fullWidth>
-                {users.map((t) => (
-                  <MenuItem key={t.userId} value={t.userId}>
-                    {fullNamePipe(t)}
-                  </MenuItem>
-                ))}
-              </TextField>
+              <>
+                <FormLabel>Project Manager</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:104]: FormControl

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 8]
```diff
@@ -17,90 +17,112 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
-            )}
-          />
+          <FormControl>
+            <Controller
+              name="projectLeadId"
+              control={control}
+              rules={{ required: true }}
+              render={({ field: { onChange, value } }) => (
+                <>
+                  <FormLabel>Project Lead</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:51]: Form Label can also go outside the controller,

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 9]
```diff
@@ -17,90 +17,112 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
+            />
+          </FormControl>
+          <FormControl>
+            <Controller
+              name="projectManagerId"
+              control={control}
+              rules={{ required: true }}
+              render={({ field: { onChange, value } }) => (
+                <>
+                  <FormLabel>Project Manager</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:70]: same as above

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 10]
```diff
@@ -21,97 +21,120 @@ const statuses = Object.values(WbsElementStatus);
...
-      )}
-    />
+    <FormControl>
+      <Controller
+        name="wbsElementStatus"
+        control={control}
+        rules={{ required: true }}
+        render={({ field: { onChange, value } }) => (
+          <>
+            <FormLabel>Status</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:31]: formlabel can go outside controller

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 11]
```diff
@@ -21,97 +21,120 @@ const statuses = Object.values(WbsElementStatus);
...
-            )}
-          />
+          <FormControl>
+            <Controller
+              name="startDate"
+              control={control}
+              rules={{ required: true }}
+              render={({ field: { onChange, value } }) => (
+                <>
+                  <FormLabel>Start Date (YYYY-MM-DD)</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:68]: same as above

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 12]
```diff
@@ -21,97 +21,120 @@ const statuses = Object.values(WbsElementStatus);
...
-            )}
-          />
+          <FormControl>
+            <Controller
+              name="projectLead"
+              control={control}
+              rules={{ required: true }}
+              render={({ field: { onChange, value } }) => (
+                <>
+                  <FormLabel>Project Lead</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:92]: same as above

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 13]
```diff
@@ -21,97 +21,120 @@ const statuses = Object.values(WbsElementStatus);
...
-            )}
-          />
+          <FormControl>
+            <Controller
+              name="projectManager"
+              control={control}
+              rules={{ required: true }}
+              render={({ field: { onChange, value } }) => (
+                <>
+                  <FormLabel>Project Manager</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:113]: same as above

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 14]
```diff
@@ -17,90 +17,112 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
       <Grid container spacing={1}>
         <Grid item xs={12} md={6} sx={{ mt: 2, mb: 1 }}>
-          <ReactHookTextField
-            name="name"
-            control={control}
-            sx={{ width: 10 / 10 }}
-            label="Project Name"
-            errorMessage={errors.name}
-          />
+          <FormControl sx={{ width: '60%' }}>
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:20]: if you set this to 80%, I think it should look fine, you can tweak that number to whatever you think looks best

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**